server/src/main/java/org/opensearch/index/translog/transfer/FileTransferTracker.java

```diff
@@ -64,10 +68,20 @@ long getTotalBytesToUpload() {
 
     @Override
     public void onSuccess(TransferFileSnapshot fileSnapshot) {
-        long durationInMillis = (System.nanoTime() - fileTransferStartTime) / 1_000_000L;
-        remoteTranslogTransferTracker.addUploadTimeInMillis(durationInMillis);
-        remoteTranslogTransferTracker.addUploadBytesSucceeded(bytesForTlogCkpFileToUpload.get(fileSnapshot.getName()));
-        add(fileSnapshot.getName(), TransferState.SUCCESS);
+        try {
+            long durationInMillis = (System.nanoTime() - fileTransferStartTime) / 1_000_000L;
+            remoteTranslogTransferTracker.addUploadTimeInMillis(durationInMillis);
+            remoteTranslogTransferTracker.addUploadBytesSucceeded(bytesForTlogCkpFileToUpload.get(fileSnapshot.getName()));
```

Comment:
Is it that fileSnapshot gets returned as null. Do we need to simply put a null check in that case?

Reply:
@Bukhtawar fileSnapshot wont be null. The speculation as of now is `bytesForTlogCkpFileToUpload` doesn't contain the entry for `fileSnapshot.getName()`, which leads to null getting assigned to a primitive long, raising NullPointerException. This happens when we get latch timeouts on translog uploads

CC: @sachinpkale 

This try catch we have added is to 1) zero down on this error and 2) to not let any failures to update stats lead to Cx requests failing.